### PR TITLE
Add Binary Ops to D2D object

### PR DESCRIPTION
**Pull request by nickbattam-tessella**
Move the binary operation manager into the DnD base class and update logic to handle DnD classes in SQW and PixelData settings.
Split binary op tests between SQW and DND test files to match use cases.

[any changes to the InferiorClasses pattern will be made under a separate PR]

### CODE REVIEW

**Comment by nickbattam-tessella** [comment]
I've had to use `classdef (InferiorClasses = {?d2d}) sqw < SQWDnDBase` in the SQW class definition to retain the dominance of the SQW object in calls to preserve the `dnd .op. sqw -> sqw` pattern.

This is a smell as the SQW class needs to include a list of the concrete DnD classes rather than this coming through the class hierarchy. Classes do not inherit their `Inferior` status so we cannot have `DnDBase` and the named class.

I'm interested in any alternate approaches people can see to this.

**Comment by abuts** [comment]
I do not understand why would we want to keep superior/inferior relationships. These should be independent classes, operated via class methods and the inferiority is something necessary for old style classes

**Comment by nickbattam-tessella** [comment]
`InferiorClass` attribute is part of the new-style class specification: https://www.mathworks.com/help/matlab/matlab_oop/class-precedence.html
The use-case we have here matches their second use-case: "If there is a dominant argument, call the method of the dominant class."
We have defined that that SQW is the dominant argument in any binary operations.

File: horace_core/sqw/PixelData/@PixelData/do_binary_op.m [hunk 1]
```diff
@@ -43,7 +43,7 @@
     pix_out = binary_op_double_(pix_out, operand, binary_op, flip, npix);
 elseif isa(operand, 'PixelData')
     pix_out = binary_op_pixels_(pix_out, operand, binary_op, flip);
-elseif ~isempty(regexp(class(operand), '^d[0-4]d_old$', 'ONCE')) || isa(operand, 'sigvar')
+elseif ~isempty(regexp(class(operand), '^d[0-4]d_old$', 'ONCE')) || isa(operand, 'DnDBase') || isa(operand, 'sigvar')
     pix_out = binary_op_sigvar_(pix_out, operand, binary_op, flip, npix);
```
Comment (abuts) [inline horace_core/sqw/PixelData/@PixelData/do_binary_op.m:47]:
```suggestion
elseif ~isempty(regexp(class(operand), '^d[0-4]d_old$', 'ONCE')) || isa(operand, 'DnDBase') || isa(operand, 'sigvar')
    pix_out = binary_op_sigvar_(pix_out, operand, binary_op, flip, npix);
 ```
Should it be `^d[0-4]d$' now?
+
Can not judge if we want binary_op on `PixelData` itself rather then doing it as part of operation on `sqw` object. But may be missing something there

**Review by abuts — APPROVED** [review]
With reference to the discussion on superiority\inferiority suitability, looks good

File: horace_core/sqw/PixelData/@PixelData/do_binary_op.m (hunk 1 above)
Comment (nickbattam-tessella) [reply horace_core/sqw/PixelData/@PixelData/do_binary_op.m:47]: The `d0d`-`d4d` (new class) case is covered by `DnDBase` check. 

In the `sqw` binary operations implementation, call made as `sqw .op. non_sqw` are implemented as 
```
sqw.pixeldata .op. non_sqw; 
recompute_image;
```
so the logic for the operations is implemented at PixelData level.

You raise a good point about why PixelData needs to know about DnD objects at though - I'll look into this.

**Review by hsautessella — APPROVED** [review]
Looks OK to me 👍

**Pull request merged**